src/test/java/org/gridsuite/modification/server/Impacts/TestImpactUtils.java

```diff
@@ -151,34 +149,29 @@ public static void testBranchImpacts(ObjectMapper mapper, SimpleImpactType impac
+    private static List<AbstractBaseImpact> createBranchImpacts(SimpleImpactType type, IdentifiableType branchType, String branchId,
                                                                  String breakerId1, String disconnectorId1, String substationId1,
                                                                  String breakerId2, String disconnectorId2, String substationId2) {
-        LinkedList<AbstractBaseImpact> impacts = new LinkedList<>(List.of(createElementImpact(impactType, branchType, branchId, new HashSet<>(List.of(substationId1, substationId2)))));
-        List<SimpleElementImpact> switchImpacts = List.of(
-            createElementImpact(impactType, IdentifiableType.SWITCH, breakerId1, Set.of(substationId1)),
-            createElementImpact(impactType, IdentifiableType.SWITCH, disconnectorId1, Set.of(substationId1)),
-            createElementImpact(impactType, IdentifiableType.SWITCH, breakerId2, Set.of(substationId2)),
-            createElementImpact(impactType, IdentifiableType.SWITCH, disconnectorId2, Set.of(substationId2))
-        );
-        if (impactType == SimpleImpactType.CREATION) {
-            // During a creation of a 2WT we use the TapChangerAdder which set the tapChangerAttributes to the 2WT
-            // This setRatioTapChanger/setPhaseTapChanger calling generates a notifyUpdate for the newly created 2WT
-            // Then we must add a MODIFICATION impact on this newly created 2WT.
-            // TODO fix this
-            impacts.add(createElementImpact(SimpleImpactType.MODIFICATION, branchType, branchId, new TreeSet<>(List.of(substationId1, substationId2)))); // case with newtapChanger
-        }
-        if (impactType == SimpleImpactType.DELETION) {
+        LinkedList<AbstractBaseImpact> impacts = new LinkedList<>();
+        if (type == SimpleImpactType.DELETION) {
+            List<SimpleElementImpact> switchImpacts = List.of(
+                createElementImpact(SimpleImpactType.DELETION, branchType, branchId, new HashSet<>(List.of(substationId1, substationId2))),
+                createElementImpact(SimpleImpactType.DELETION, IdentifiableType.SWITCH, breakerId1, Set.of(substationId1)),
+                createElementImpact(SimpleImpactType.DELETION, IdentifiableType.SWITCH, disconnectorId1, Set.of(substationId1)),
+                createElementImpact(SimpleImpactType.DELETION, IdentifiableType.SWITCH, breakerId2, Set.of(substationId2)),
+                createElementImpact(SimpleImpactType.DELETION, IdentifiableType.SWITCH, disconnectorId2, Set.of(substationId2))
+            );
             impacts.addAll(0, switchImpacts);
         } else {
-            impacts.addAll(switchImpacts);
+            Set<String> substationIds = new HashSet<>();
+            substationIds.addAll(List.of(substationId1, substationId2));
+            impacts.addAll(createSubstationImpacts(substationIds));
```

Comment:
Why not Set.of() here directly?

Reply:
because : https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Set.html#of(E...)
throw : `IllegalArgumentException` - if there are any duplicate elements

Reply:
I will use Set.copyOf